html/includes/authentication/functions.php

```diff
@@ -142,9 +142,12 @@ function set_remember_me()
         dbInsert($db_entry, 'session');
     }
 
-    setcookie('sess_id', $sess_id, $expiration, '/', null, false, true);
-    setcookie('token', $token_id, $expiration, '/', null, false, true);
-    setcookie('auth', $auth, $expiration, '/', null, false, true);
+    // Params: name, value, expire, path, domain, secure, httponly
+    // We set the secure param to $_SERVER["HTTPS"], which will return true
+    // and make it a secure cookie if HTTPS is being used
+    setcookie('sess_id', $sess_id, $expiration, '/', null, $_SERVER["HTTPS"], true);
```

Comment:
Currently secure cookies are turned off regardlesss of whether the user is connecting over HTTP or HTTPS. I figured having it done automatically would be the best solution, but can change it to be in `config.php` if you want.